### ISSUE TITLE
Can't cast address to u256

### ISSUE THREAD

**Issue by sbillig** [ethereum/fe#614]
### What is wrong?

There's no way to get the `u256` value of an `address` (as far as I can tell).

```
fn f(addr: address) -> u256:
  return u256(addr)

error: type mismatch
  ┌─ /home/sean/code/eth/address_cast.fe:2:15
  │
2 │   return u256(addr)
  │               ^^^^ expected a numeric type but was `address`
```

### How can it be fixed

Change some things around analyzer/src/traversal/expressions.rs:1273

**Comment by sbillig** [comment]
I should add that this is needed to provide nice(r) wrappers around low-level intrinsics. For example:

```
pub unsafe fn balance_of(addr: address) -> u256:
  unsafe:
    return __balance(u256(addr)) # all intrinsics take u256
```

**Comment by sbillig** [comment]
fixed in #621
